packages/langium/src/grammar/langium-grammar-validator.ts

```diff
@@ -308,6 +309,16 @@ export class LangiumGrammarValidator {
         }
     }
 
+    checkFragmentsInTypes(grammar: ast.Grammar, accept: ValidationAcceptor): void {
+        grammar.types.forEach(e => {
+            e.typeAlternatives.forEach(e => {
+                if (ast.isParserRule(e.refType?.ref) && (e.refType?.ref as ast.ParserRule).fragment) {
```

Comment:
The type cast via `as` is not necessary here because it has already been checked with `isParserRule`.